sklearn/inspection/_permutation_importance.py

```diff
@@ -143,14 +159,36 @@ def permutation_importance(estimator, X, y, *, scoring=None, n_repeats=5,
     random_state = check_random_state(random_state)
     random_seed = random_state.randint(np.iinfo(np.int32).max + 1)
 
-    scorer = check_scoring(estimator, scoring=scoring)
-    baseline_score = _weights_scorer(scorer, estimator, X, y, sample_weight)
-
-    scores = Parallel(n_jobs=n_jobs)(delayed(_calculate_permutation_scores)(
-        estimator, X, y, sample_weight, col_idx, random_seed, n_repeats, scorer
-    ) for col_idx in range(X.shape[1]))
-
-    importances = baseline_score - np.array(scores)
-    return Bunch(importances_mean=np.mean(importances, axis=1),
-                 importances_std=np.std(importances, axis=1),
-                 importances=importances)
+    if scoring is None or isinstance(scoring, str) or callable(scoring):
+        scorer = check_scoring(estimator, scoring=scoring)
+        baseline_score = _weights_scorer(scorer, estimator, X, y,
+                                         sample_weight)
+
+        scores = Parallel(n_jobs=n_jobs)(
+            delayed(_calculate_permutation_scores)(
+                estimator, X, y, sample_weight, col_idx, random_seed,
+                n_repeats, scorer
+            ) for col_idx in range(X.shape[1]))
+
+        importances = baseline_score - np.array(scores)
+        return Bunch(importances_mean=np.mean(importances, axis=1),
+                     importances_std=np.std(importances, axis=1),
+                     importances=importances)
+    else:
+        scorers = _check_multimetric_scoring(estimator, scoring)
+        ret = dict()
+        for name, scorer in scorers.items():
+            baseline_score = _weights_scorer(scorer, estimator, X, y,
```

Comment:
I suppose that this call to scorer will cache the prediction.

Reply:
Indeed that's right, the cache should be initialized in the sequential call for the baseline and then passed onto the workers.

We might have a problem with communication overhead though but I don't see a way around.